include/ck/tensor_operation/gpu/device/device_layernorm.hpp

```diff
@@ -106,47 +142,49 @@ struct DeviceLayernorm : public BaseOperator
-                                                                      MThreadClusterSize,
-                                                                      KThreadClusterSize,
-                                                                      MThreadSliceSize,
-                                                                      KThreadSliceSize,
-                                                                      XYSrcVectorDim,
-                                                                      XSrcVectorSize,
-                                                                      GammaSrcVectorSize,
-                                                                      BetaSrcVectorSize,
-                                                                      XYSrcVectorDim,
-                                                                      YDstVectorSize,
-                                                                      false>;
-
-    using GridwiseReduceLayernormSweepOnce = GridwiseLayernorm_mk_to_mk<XDataType,
-                                                                        GammaDataType,
-                                                                        BetaDataType,
-                                                                        YDataType,
-                                                                        AccDataType,
-                                                                        AccElementwiseOperation,
-                                                                        GridDesc_M_K,
-                                                                        GridDesc_K,
-                                                                        BlockSize,
-                                                                        MThreadClusterSize,
-                                                                        KThreadClusterSize,
-                                                                        MThreadSliceSize,
-                                                                        KThreadSliceSize,
-                                                                        XYSrcVectorDim,
-                                                                        XSrcVectorSize,
-                                                                        GammaSrcVectorSize,
-                                                                        BetaSrcVectorSize,
-                                                                        XYSrcVectorDim,
-                                                                        YDstVectorSize,
-                                                                        true>;
+    using GridwiseReduceLayernormGeneric =
+        GridwiseLayernormWelfordVariance_mk_to_mk<XDataType,
+                                                  GammaDataType,
+                                                  BetaDataType,
```

Comment:
`gamma` and `beta` are model parameters,  they should have the same type

Reply:
Distribution of gamma and beta can be different.
Consider the case of mix precision, if we want to take a good balance between bandwidth and precision, type of each model parameter might be different.